src/thor/bidirectional_astar.cc

```diff
@@ -916,10 +918,167 @@ void BidirectionalAStar::SetDestination(GraphReader& graphreader, const valhalla
   }
 }
 
+/**
+ * Recover shortcut edge and recost it
+ * @param  graphreader  Graph tile reader.
+ * @param  shortcut_id  Edge id of the shortcut to be recosted
+ * @param  costing      Costing function
+ * @param  transition_cost  Transition cost for the shortcut edge
+ * @param  previous_cost    Total cost from the previous edge
+ * TODO: add time information
+ */
+std::vector<PathInfo> recost_shortcut_forward(GraphReader& graphreader,
```

Comment:
Is recosting shortcuts something that could benefit all A-star algorithms? If so, could this be implemented on the base-class?

Reply:
I think yes, will check

Reply:
heavy plus for this one

Reply:
all A-start algorithms inherit from `PathAlgorithm` interface. as for me, it's too common in order to add `recost_shortcut_forward` to this interface. instead of that, I will define this function in a separate *.h file (in `src/thor` folder) that can be used by everyone